java-squid/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -201,10 +202,17 @@ private JavaSymbol findMemberType(Env env, JavaSymbol.TypeJavaSymbol site, Strin
         bestSoFar = symbol;
       }
     }
-    for (JavaType interfaceType : c.getInterfaces()) {
-      JavaSymbol symbol = findMemberType(env, site, name, interfaceType.symbol);
-      if (symbol.kind < bestSoFar.kind) {
-        bestSoFar = symbol;
+    if (c.getInterfaces() == null) {
+      // Invariant to check that interfaces are not set only when we are looking into the symbol we are currently completing.
+      // required for generics
+      Preconditions.checkState(c.completing, "interfaces of a symbol not currently completing are not set.");
+      Preconditions.checkState(c == site);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)